frontend/public/module/k8s/types.ts

```diff
@@ -483,25 +483,29 @@ export type CRDVersion = {
   name: string;
   served: boolean;
   storage: boolean;
+  schema?: {
+    // NOTE: Actually a subset of JSONSchema, but using this type for convenience
+    openAPIV3Schema: JSONSchema6;
+  };
 };
 
 export type CustomResourceDefinitionKind = {
   spec: {
-    version: string;
     group: string;
+    version?: string;
+    versions?: CRDVersion[];
     names: {
       kind: string;
       singular: string;
       plural: string;
       listKind: string;
       shortNames?: string[];
     };
-    scope?: 'Namespaced';
+    scope: 'Cluster' | 'Namespaced';
     validation?: {
       // NOTE: Actually a subset of JSONSchema, but using this type for convenience
       openAPIV3Schema: JSONSchema6;
     };
-    versions?: CRDVersion[];
   };
```

Comment:
Might need some input from @spadgett on this. This type is structured to support both v1 and v1beta1 because they are both supported in Kubernetes v1.18, but I'm not sure if that's necessary or not.